provisioner/aws-manager.js

```diff
@@ -129,6 +134,32 @@ AwsManager.prototype._filterSpotRequests = function(spotReqs) {
   return data;
 };
 
+
+/**
+ * Instead of defining these lists in lots of places, let's ensure
+ * that we use the same filters in all places!
+ */
+
+AwsManager.prototype.filters = {
```

Comment:
So as I understand this is a the list of properties on objects representing spot requests and instances, respectively... Is that correct?

One of the things I've been missing was a well-defined representation of these objects. This is definitely a decent solution to that :)
If we go down this road we should probably annotate which of these properties are optional.

I was thinking of doing this with classes, example:
```js
class Instance {
  constructor(resultFromEC2) {
    this.__raw = resultFromEC2
  }
  get instanceId() {
    return this.__raw.InstanceId;
  }
  get availabilityZone() {
    if (!this.__raw.LaunchSpecification ||
        this.__raw.LaunchSpecification.Placement) {
      return undefined;  
    }
    return this.__raw.LaunchSpecification.Placement.AvailabilityZone;
  }
}
```
This way we could flatten the objects and make the properties more accessible. Hide some of the stupid logic that avoids issues when `LaunchSpecification.Placement` is missing. Note this was just a illustrative example, I'm not sure this pattern is better... It certainly adds a lot of unnecessary code.

**Main point** is let's have something that describes what properties the objects have. So we can quickly look it up. We don't have anything like that prior to this PR, and we shouldn't trust AWS docs :)

----
Any reason why these should be accessible outside the class?
Are they not internal.. static? Ie. how about a global constant `const spotReqFilter = [...]`.